libparsec/crates/types/tests/unit/id.rs

```diff
@@ -38,3 +38,20 @@ fn human_handle(#[case] email: &str, #[case] label: &str, #[case] is_ok: bool) {
         is_ok
     );
 }
+
+#[rstest]
+#[case::alphanum("01239AbcdDEFA", true)]
+#[case::with_underscore("hello_World", true)]
+#[case::with_hyphen("hello-World", true)]
+// cspell:disable-next-line
+#[case::to_long("aaaabbbbccccddddeeeeffffgggghhhhiiiijjjjj", false)]
+// cspell:disable-next-line
+#[case::with_unicode_1("ªµºÖØøˆˠˬˮ\u{301}ͶͽͿΆΊ", true)]
+#[case::with_unicode_2("ΌΎΣҁ\u{484}Աՙֈ\u{5af}\u{5bf}\u{5c1}\u{5c4}\u{5c7}", true)]
+#[case::with_unicode_3("ת\u{5ef}\u{610}٩ۓە", true)]
+#[case::with_unicode_4("\u{6df}\u{6ea}ۿܐݍ߀ߺ\u{7fd}ࠀࡀ", true)]
+// cspell:disable-next-line
+#[case::hello("𝔅𝔬𝔫𝔧𝔬𝔲𝔯", true)]
+fn test_match_legacy_id(#[case] input: &str, #[case] result: bool) {
```

Comment:
I'd rather merge this with `libparsec/crates/types/tests/id.rs's `organization_id_user_id_and_device_name`/`bad_organization_id_user_id_and_device_name` tests as it is better to test the actual components we care about that they internals

Reply:
done